src/modules/custom.rs

```diff
@@ -160,6 +160,11 @@ fn shell_command(cmd: &str, shell_args: &[&str]) -> Option<Output> {
             .stdout(Stdio::piped())
             .stderr(Stdio::piped());
 
+        if forced_shell.ends_with("cmd.exe") || forced_shell.ends_with("cmd") {
+            log::error!("Custom command output with Command shell is not supported.");
```

Comment:
I would prefer calling it `cmd` and I think this should include a troubleshooting suggestion.

```suggestion
            log::error!("Executing custom commands with the cmd shell is not currently supported. Please set a different shell with the \"shell\" option.");
```

Reply:
Although I have made the requested change here, there's one important thing to note: the `when` condition of custom command works correctly with cmd shell. It's just that the `command` field (whose output gets shown actually) is problematic. This is why I mentioned _Custom command output_ in my original error message.

Reply:
The log message is going to annoy them either way, so I'm not sure how helpful rewording this will be. Unless this can be restricted not to show up for `when` maybe this should just become an `info` log. Does anyone else have any ideas?

Reply:
This brings me to a question: when we specify shell explicitly through the `shell` option, is that explicit shell used both for `when` and `command`, or just for `command`? 

If it's the former, then I guess we don't really have a choice and we can keep the current message. 

However, if it's the latter (where `when` will always be executed through cmd.exe, and `command` will get executed through the explicitly specified `shell`), then we will need to be more verbose here as to what is actually not allowed/supported.

Reply:
They both use `shell`

Reply:
Then I think this is good to go

Reply:
In that case, can you move the message into [the `"output" =>` part of `map_no_escaping`](https://github.com/starship/starship/blob/808ddfca7a067b6625a4fe2869a36fcbb78ad9e7/src/modules/custom.rs#L60-L69) and gate it with `context.shell == Shell::Cmd && config.shell == None`?

Reply:
Done